src/osp/Active/ActiveScene.cpp

```diff
@@ -216,25 +216,33 @@ void ActiveScene::update_hierarchy_transforms()
             // set transform relative to parent
             transform.m_transformWorld = parentTransform.m_transformWorld
                                           * transform.m_transform;
-
         }
     }
-
 }
 
 void ActiveScene::draw(ActiveEnt camera)
 {
-    //Matrix4 cameraProject;
-    //Matrix4 cameraInverse;
-
-    // TODO: check if camera has the right components
-    ACompCamera& cameraComp = reg_get<ACompCamera>(camera);
-    ACompTransform& cameraTransform = reg_get<ACompTransform>(camera);
-
-    //cameraProject = cameraComp.m_projection;
-    cameraComp.m_inverse = cameraTransform.m_transformWorld.inverted();
+    auto renderView = m_registry.view<ACompRenderingAgent, ACompPerspective3DView, ACompRenderer>();
+    for (auto [e, agent, view, renderer] : renderView.each())

```

Comment:
imho, this would be better as `auto const&`, but whether that actually results in different code gen depends on the specific types of the variables.

Reply:
This doesn't really seem to change the types that get used, which are `T&&` for the entity type (`e`) and `T&` for all the component types. This doesn't change even if I add `const`. I'm going to leave it as-is since this is the way skypjack himself seems to write it.